miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/controller/rest/SampleRestController.java

```diff
@@ -263,28 +269,65 @@ public SampleDto updateSample(@PathVariable("id") Long id, @RequestBody SampleDt
 
   /**
    * 
-   * @param identitiesSearches
+   * @param json
    *          String externalNames or Identity aliases
    * @param response
    * @return
    * @throws IOException
    */
   @RequestMapping(value = "/identities", method = RequestMethod.POST, headers = { "Content-type=application/json" })
   @ResponseBody
-  public Set<SampleDto> getIdentitiesBySearch(@RequestBody com.fasterxml.jackson.databind.JsonNode json,
+  public Set<SampleDto> getIdentityBySearch(@RequestBody com.fasterxml.jackson.databind.JsonNode json,
       HttpServletResponse response) throws IOException {
-    if (json.get("identitiesSearches").asText().length() == 0) {
-      throw new RestException("Must give search terms to look up identities", Status.BAD_REQUEST);
+    final JsonNode searchTerms = json.get("identitiesSearches");
+    final String project = (json.get("project") == null ? "" : json.get("project").asText());
+    if (!searchTerms.isArray() || searchTerms.size() == 0) {
+      throw new RestException("Please provide external name or alias for identity lookup", Status.BAD_REQUEST);
     }
-    Set<Sample> uniqueIdentities = new HashSet<>();
-    String searchTerms = json.get("identitiesSearches").asText();
-    for (String term : SampleIdentityImpl.getSetFromString(searchTerms.replaceAll(";", ","))) {
-      Collection<SampleIdentity> matches = sampleService.getIdentitiesByExternalNameOrAlias(term);
-      for (SampleIdentity identity : matches) {
-        uniqueIdentities.add(identity);
+    return getSamplesForIdentityString(searchTerms.get(0).asText(), project, true);
+  }
+
+  /**
+   * Only returns identities which exactly match
+   * 
+   * @param json
+   * @param response
+   * @return
+   * @throws IOException
+   */
+  @RequestMapping(value = "/identitiesLookup", method = RequestMethod.POST, headers = { "Content-type=application/json" })
+  public @ResponseBody List<Map<String, Set<SampleDto>>> getIdentitiesBySearch(@RequestBody com.fasterxml.jackson.databind.JsonNode json,
+      HttpServletResponse response) throws IOException {
+    final JsonNode searchTerms = json.get("identitiesSearches");
+    final String project = (json.get("project") == null ? "" : json.get("project").asText());
+    if (!searchTerms.isArray() || searchTerms.size() == 0) {
+      throw new RestException("Please provide external name or alias for identity lookup", Status.BAD_REQUEST);
+    }
+    List<Map<String, Set<SampleDto>>> identitiesBySearchTerm = new ArrayList<>();
+    for (JsonNode term : searchTerms) {
+      Set<SampleDto> uniqueIdentities = getSamplesForIdentityString(term.asText(), project, false);
+      if (uniqueIdentities.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)